### PR TITLE
Perform UCR sorting via sqlagg

### PR DESCRIPTION
**Pull request by nickpell**
@czue @snopoke @NoahCarnahan cc: @sravfeyn 

TODO:
- [x] package https://github.com/dimagi/sql-agg/pull/28
- [x] Add tests

### CODE REVIEW

**Comment by snopoke** [comment]
There are a few other usages of QueryMeta that will also need to be updated.

**Comment by nickpell** [comment]
@czue @NoahCarnahan 

The test failure

```
======================================================================
ERROR: test_aggregation_by_indicator_in_percent_column (corehq.apps.userreports.tests.test_report_aggregation.TestReportAggregation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/userreports/tests/test_report_aggregation.py", line 182, in test_aggregation_by_indicator_in_percent_column
    view.export_table,
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 128, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/corehq/apps/userreports/reports/view.py", line 358, in export_table
    raw_rows = list(data.get_data())
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 128, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/corehq/apps/userreports/reports/data_source.py", line 143, in get_data
    raise UserReportsError(unicode(e))
UserReportsError: (psycopg2.ProgrammingError) column "report_column_percent" does not exist
LINE 2: ...IN_foo_78856a50".indicator_col_id_number ORDER BY report_col...
                                                             ^
 [SQL: 'SELECT sum("config_report_TEST_DOMAIN_foo_78856a50".indicator_col_id_number) AS indicator_col_id_number, sum("config_report_TEST_DOMAIN_foo_78856a50".indicator_col_id_number) AS report_column_numerator \nFROM "config_report_TEST_DOMAIN_foo_78856a50" GROUP BY "config_report_TEST_DOMAIN_foo_78856a50".indicator_col_id_number ORDER BY report_column_percent ASC']
```

reveals a problem with sorting in the database - it's not clear how to handle percentage columns.  Would you have to have nested selects to do the division in SQL?

https://github.com/dimagi/commcare-hq/commit/aba0a8254e71075299436b4e301f7acf9b6bef7d likewise reveals that we can't sort on the output of a transform of a report column using DB sorting.  However you can workaround by applying the transform to the data source indicators, so I don't consider it to be a big deal.

**Comment by czue** [comment]
I think the option you're proposing of pushing it to the DB would mean that percentage columns would have to be fully pushed to the DB, yeah? That seems like a good change though potentially out of scope.

do you think it would be reasonable to disable sorting by percentage as a medium term answer? seems like it could be more trouble than it's worth.

**Comment by nickpell** [comment]
@czue I'm in favor of punting it.  On prod there are only 3 reports that sort by a percentage column, and only one of them has more than one row - https://www.commcarehq.org/a/squash/reports/configurable/f7e8e0ebeea2d1e271bd5603f2d1e359/ - and it sorts alphabetically anyway.

**Comment by czue** [comment]
:+1: to :boot:

File: corehq/apps/userreports/tests/test_report_aggregation.py [hunk 1]
```diff
@@ -229,3 +229,64 @@ def test_aggregation_by_column_with_new_id(self):
...
+                ]
+            ]]
+        )
+
+        report_config.sort_expression = [{
+            'field': 'indicator_col_id_first_name',
+            'order': 'DESC',
+        }]
+        report_config.save()
+        descending_sorted_view = self._create_view(report_config)
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_report_aggregation.py]: F841 local variable 'descending_sorted_view' is assigned to but never used

**Comment by nickpell** [comment]
:white_check_mark:

**Comment by czue** [comment]
:boom:

**Pull request merged**